tests/phpunit/tests/image/functions.php

```diff
@@ -700,4 +700,106 @@ public function test_pdf_preview_doesnt_overwrite_existing_jpeg() {
 			unlink( $temp_dir . $size['file'] );
 		}
 	}
+
+	/**
+	 * Test for wp_exif_frac2dec verified that it properly handles edge cases
+	 * and always returns an int or float, or original value for failures.
+	 *
+	 * @param string $fraction The fraction to convert.
+	 * @param string $expect   The expected result.
+	 *
+	 * @ticket 54385
+	 * @dataProvider data_wp_exif_frac2dec
+	 *
+	 * @covers ::wp_exif_frac2dec
+	 */
+	public function test_wp_exif_frac2dec( $fraction, $expect ) {
+		$this->assertSame( $expect, wp_exif_frac2dec( $fraction ) );
+	}
+
+
+	/**
+	 * Data provider for testing `wp_exif_frac2dec()`.
+	 *
+	 * @return array {
+	 *     Arguments for testing `wp_exif_frac2dec()`.
+	 */
+	public function data_wp_exif_frac2dec() {
+		global $wpdb;
+
+		return array(
+			'division by zero is prevented'       => array(
+				'fraction' => '0/0',
+				'expect'   => '0/0',
+			),
+			'typical fnumber'                     => array(
+				'fraction' => '4.8',
+				'expect'   => '4.8',
+			),
+			'typical focal length'                => array(
+				'fraction' => '37 mm',
+				'expect'   => '37 mm',
+			),
+			'typical exposure time'               => array(
+				'fraction' => '1/350',
+				'expect'   => 0.002857142857142857,
+			),
+			'division by text prevented'          => array(
+				'fraction' => '0/abc',
+				'expect'   => '0/abc',
+			),
+			'non fraction returns original value' => array(
+				'fraction' => '0.0',
+				'expect'   => '0.0',
+			),
+			'non fraction returns original value' => array(
+				'fraction' => '010',
+				'expect'   => '010',
+			),
+			'non fraction returns original value' => array(
+				'fraction' => 10.123,
+				'expect'   => 10.123,
+			),
```

Comment:
Only the last of these test cases will currently run as you're using the exact same (duplicate) array key.